impeller/renderer/backend/vulkan/capabilities_vk.cc

```diff
@@ -351,10 +351,8 @@ bool CapabilitiesVK::SetPhysicalDevice(const vk::PhysicalDevice& device) {
 
   if (HasSuitableDepthStencilFormat(device, vk::Format::eS8Uint)) {
     default_stencil_format_ = PixelFormat::kS8UInt;
-  } else if (default_stencil_format_ != PixelFormat::kUnknown) {
-    default_stencil_format_ = default_depth_stencil_format_;
   } else {
```

Comment:
This will set the `default_depth_stencil_format_` to `kUnknown` after we may have already initialized it above. Instead maybe we can just delete this entire block?

Reply:
wait no, that is default_depth_stencil_format_ which is different. woops

Reply:
I deleted this since it was checking against `default_stencil_format_` and not `default_depth_stencil_format_`, and so it would always end up being `PixelFormat::kUnknown` because we haven't set it yet. I think that was a logic error.

Either way, making the originally intended fallback work SGTM. Done.